Adds methods to convert.py to support data compression/decompression

theJokerEvoker opened the pull request:
# Overview

Adds methods to convert.py to support data compression/decompression.

## Ticket

https://www.pivotaltracker.com/story/show/179306557

## Contributions

- Modified convert.py, added compress, decompress, archive_list, tar_name_checker
- compress - Adds files to a tar archive and compresses them using the gzip compression format.
- decompress - Takes .tar.gz archive, decompresses, and extracts its contents. Should only be used with files created by compress() method.
- archive_list - Takes .tar.gz archive and returns a full list of its contents, including folders, subfolders, and files, with their full relative paths and names.
- tar_name_checker - Helper method that takes a tar file name and checks it for the appropriate file extension (".tar.gz" for now), returning it if it already does, and adding the extension and then returning it if it does not.

## Test

- Run test_convert.py unit test suite
- Tested successfully on BciPy data recording, compressed 686 MB folder to 572 MB archive (17% compression)

## Documentation

- Added docstrings where applicable

Commit:
Adds methods to convert.py to support data compression/decompression

Commit:
Linting

tab-cmd commented on bcipy/helpers/convert.py:
```diff
@@ -2,18 +2,22 @@
 
 logger = logging.getLogger(__name__)
 
+FILE_LENGTH_LIMIT = 150
```
This could be an input argument with a default, but this seems okay too since it's a warning

tab-cmd commented on bcipy/helpers/convert.py:
```diff
@@ -210,3 +214,79 @@ def edf_annotations(triggers: List[Tuple[str, str, float]],
+
+
+def compress(tar_file_name: str, members: List[str]) -> None:
+    """Compress.
```
Update docstrings to match other methods

tab-cmd commented on bcipy/helpers/convert.py:
```diff
@@ -210,3 +214,79 @@ def edf_annotations(triggers: List[Tuple[str, str, float]],
+
+
+def decompress(tar_file: str, path: str) -> None:
+    """Extracts tar_file and places specified "members" in "path"
```
Same as above

tab-cmd reviewed (approved):
Consider suggestions and update docstings

lawhead commented on bcipy/helpers/convert.py:
```diff
@@ -210,3 +214,79 @@ def edf_annotations(triggers: List[Tuple[str, str, float]],
+       If members remains None, all members will be extracted
+    """
+
+    full_tar_name = f'{tar_file}.tar.gz'
```
It's not clear from the method documentation that you only want the base name of the file. You should either update the documentation or make the implementation more lenient by checking for the suffix (or stripping it) before adding it here. The same comment applies to the other methods. Given the repetition, that code could be factored out into another function.

Commit:
Added documentation, added name_checker helper method, added unit tests

Added tar_name_checker helper method to check for properly named tar archives with extensions, and added appropriate unit tests to check that it works within the three existing compression methods. Added documentation for the four compression related methods. Renamed file_list to archive_list.

tab-cmd committed:
Small changes to merge PR
